File: Examples/test-suite/ruby/newobject1_runme.rb
require 'newobject1'

include Newobject1

foo1 = Foo.makeFoo
raise RuntimeError if Foo.fooCount != 1

foo2 = foo1.makeMore
raise RuntimeError if Foo.fooCount != 2

foo1 = nil ; GC.start
raise RuntimeError if Foo.fooCount != 1

foo2 = nil ; GC.start
raise RuntimeError if Foo.fooCount != 0

File: Examples/test-suite/lua/newobject1_runme.lua
require("import")	-- the import fn
import("newobject1")	-- import code

foo1 = newobject1.Foo_makeFoo()	-- lua doesnt yet support static fns properly
assert(newobject1.Foo_fooCount() == 1)	-- lua doesnt yet support static fns properly

foo2 = foo1:makeMore()
assert(newobject1.Foo_fooCount() == 2)

foo1 = nil 
collectgarbage()
assert(newobject1.Foo_fooCount() == 1)

foo2 = nil 
collectgarbage()
assert(newobject1.Foo_fooCount() == 0)
